### ISSUE TITLE
Error material afx/black & lights & smoke when using afx/wh

### ISSUE THREAD

**Issue by DuKeM-CSGO** [ripieces/advancedfx#778]
I decided to record two streams in CS:GO:
```
] mirv_streams print
index: name -> recorded?, preview?
0: afxWh -> RECORD ON (1), no preview
1: nor -> RECORD ON (1), no preview  //Normal Stream
=== Total streams: 2 ===
```
And just like those msgs, I didn't preview any stream.
But when I executed the command `mirv_streams record start; demo_resume`, there're error msgs in the console:
```
] mirv_streams record start; demo_resume
Starting recording ... done.
Recording to "E:\Gustab Munchausen\CSGO Movie\CommunityServers\FFMPEG\take0008".
Material afx/black does not support vertex format used by the mesh (maybe missing fields or mismatched vertex compression?), mesh will not be rendered. Grab a programmer!
Material afx/black does not support vertex format used by the mesh (maybe missing fields or mismatched vertex compression?), mesh will not be rendered. Grab a programmer!
```
(However I didn't think the material, using `UnlitGeneric`, could cause those errors.)
With those material errors, there're two visual error in the game:
### Smokes
Have to say it's a so covert bug that I've not realized before a guy reporting. When the camera goes through a smoke and touches its edge, all the world disappeared except the skybox.
![SmokeBug](https://user-images.githubusercontent.com/61344402/175769666-ab47b6a5-7083-405e-9f78-69d552f5a546.png)
(Actually I was next to the A site and it was already round 2. It looked like it's when the demo just played)
And this frame was add into my `Normal` video, which is found by Premiere. I tried adding only one of `Normal` or `afxWh` and recording in the same way, but there's not a error frame anymore.

### Lights
The light on the _library_'s celling was not blocked right. This is a continuing bug and I met it some years ago, on _de_train_.
![LightBug](https://user-images.githubusercontent.com/61344402/175769501-be7266bf-48a7-4c9c-a7cd-82cda6fccf0b.jpg)

**Comment by dtugend** [comment]
## Smoke
The 1-3 frames skybox only when unfading is a CS:GO bug that was introduced after they fixed the mirage smoke exploit a few weeks ago during the CS:GO Major, not an HLAE bug:
https://twitter.com/dtugend/status/1526055816394047489
Feel free to report it to Valve / csgoteamfeedback@valvesoftware.com
Not 100% sure if it will work, but you can try to work around this bug by setting:
```
mirv_streams edit nor smokeOverlayAlphaFactor 0.999
```
## Lights
by default HLAE will make CS:GO use the first stream to determine the visiblity of things and HLAE will tell it to freeze it for other streams rendered after that.
This means the order in which you add streams is important.
Make sure your normal stream is first or move it to be first:
```
mirv_streams move nor 0
```

**Comment by dtugend** [comment]
I am assuming this was answered until futher notice :)

**Comment by DuKeM-CSGO** [comment]
Sorry for forgetting to test this, I'd just like to say it works without any error frames.
However, to be honest it looked more terrible when the movie is recording. With no streams to preview(as the following setting), when the camera entered the smoke totally, my CS:GO just left the skybox alone, until the camera exited the smoke.
Then I checked the `normal stream` video by Premiere, and I didn't found any bug frames in it.
```
index: name -> recorded?, preview?
0: nor -> RECORD ON (1), no preview
1: afxWh -> RECORD ON (1), no preview
=== Total streams: 2 ===
```
Anyway thx for your help and it looks solved:D

(By the way, I just tried to build hlae _again_ and now it's successful with pleasure.)
